src/containers/Secrets/Secrets.js

```diff
@@ -66,16 +66,18 @@ export /* istanbul ignore next */ class Secrets extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    const { filters, namespace, webSocketConnected } = this.props;
+    const { filters, namespace, webSocketConnected, secrets } = this.props;
     const {
       filters: prevFilters,
       namespace: prevNamespace,
-      webSocketConnected: prevWebSocketConnected
+      webSocketConnected: prevWebSocketConnected,
+      secrets: prevSecrets
     } = prevProps;
 
     if (
       !isEqual(filters, prevFilters) ||
       namespace !== prevNamespace ||
+      !isEqual(secrets, prevSecrets) ||
```

Comment:
seems a little odd, if the secrets have changed, call the API again...

is the problem that we're only extracting a subset of data from the secret and missing some fields, or are we actually receiving an incomplete object over the websocket or something?

Reply:
the problem is that the `createSecret/deleteSecret`  functions haven't always finished executing before this gets called, and so we were initially receiving fragments of secrets half way through their creation and that was what was being put in the table (or in the delete case they would stay in the table after being deleted). With this change it checks for when the secret is finished creating/deleted  then refreshes to show this in the table.

Reply:
Can we not wait for the API calls to complete?

Reply:
That was our initial attempt, @Megan-Wright can go into more details and provide clarifications - i believe we've got some nice "lifecycle/flow" logs to share, and fetchData's `secrets` weren't *full* in terms of having all the fields (type and username were particularly problematic)

Reply:
@AlanGreene yeah we tried this but the SecretsModal is the one doing `createSecret` and this works completely independently of the Secrets page.... I basically ended up sticking an`await` before every single function within secret creation and the Secrets page would still refresh in the background...

Reply:
I think a lot of the problems we're having with secrets stem from the fact that we have inconsistent treatment of them in different parts of the app. We're not storing the full kubernetes resource, instead using a custom structure with a subset of the fields, and we end up with slight differences depending on how the secrets are loaded (i.e. created via the UI, received over websocket, or via API call)

Here's an attempt at storing the full kubernetes resource as we do with other types, eliminating the custom object. https://github.com/AlanGreene/dashboard/commit/8fe96da2ecd2d5372ac97545afd9bac39ac4b98a

I think it would make sense to deliver something like that, and then build on it to resolve the loading issues you're seeing. We really shouldn't need to re-request the secrets in componentDidUpdate and there's a chance it could get called multiple times until the stored secrets stabilise.

Was the problem only reproduced on OpenShift or have you seen it on other platforms too? (e.g. docker desktop). If we handle the errors / incomplete data does it eventually load correctly or does it stay in a partially loaded state?

I'll take a look and see if I can figure out why things aren't behaving.

Reply:
> I think a lot of the problems we're having with secrets stem from the fact that we have inconsistent treatment of them in different parts of the app. We're not storing the full kubernetes resource, instead using a custom structure with a subset of the fields, and we end up with slight differences depending on how the secrets are loaded (i.e. created via the UI, received over websocket, or via API call)
> 
> Here's an attempt at storing the full kubernetes resource as we do with other types, eliminating the custom object. [AlanGreene@8fe96da](https://github.com/AlanGreene/dashboard/commit/8fe96da2ecd2d5372ac97545afd9bac39ac4b98a)
> 
> I think it would make sense to deliver something like that, and then build on it to resolve the loading issues you're seeing. We really shouldn't need to re-request the secrets in componentDidUpdate and there's a chance it could get called multiple times until the stored secrets stabilise.
> 
> Was the problem only reproduced on OpenShift or have you seen it on other platforms too? (e.g. docker desktop). If we handle the errors / incomplete data does it eventually load correctly or does it stay in a partially loaded state?
> 
> I'll take a look and see if I can figure out why things aren't behaving.

Yeah, big +1 on this, let's look into this as part of the 0.5.1 milestone, we saw this most reliably on OpenShift (the secrets page itself takes several seconds to render in this particular environment), we'll get back to you with regards to Docker Desktop

Reply:
PR based on my initial testing mentioned in the comment above: https://github.com/tektoncd/dashboard/pull/1042

This should hopefully simplify things a bit, and make it easier to get to the bottom of any remaining issues.